File: api/schemas.py
from pydantic import BaseModel


class PredictRequest(BaseModel):
    ip: str
    timestamp: float
    endpoint: str
    method: str
    status_code: int
    payload_size: float
    user_agent: str
    response_time: float


class PredictResponse(BaseModel):
    ip: str
    is_anomaly: bool
    anomaly_score: float
    top_features: dict[str, float]
    timing_ms: dict[str, float]

File: ultra-low-latency/server/src/schemas.rs
use serde::{Deserialize, Serialize};
use std::collections::HashMap;

#[derive(Deserialize)]
pub struct PredictRequest {
    pub ip: String,
    pub timestamp: f64,
    pub endpoint: String,
    pub method: String,
    pub status_code: i32,
    pub payload_size: f64,
    pub user_agent: String,
    pub response_time: f64,
}

#[derive(Serialize)]
pub struct PredictResponse {
    pub ip: String,
    pub is_anomaly: bool,
    pub anomaly_score: f64,
    pub top_features: HashMap<String, f64>,
    pub timing_ms: HashMap<String, f64>,
}
